tests/bsim/bluetooth/host/att/long_read/main.c

```diff
@@ -0,0 +1,232 @@
+BUILD_ASSERT(__ASSERT_ON);
+
+#define CENTRAL_DEVICE_NBR 0
+#define PERIPHERAL_DEVICE_NBR 1
+
+LOG_MODULE_REGISTER(main, LOG_LEVEL_DBG);
+
+#define UUID_1                                                                                     \
+	BT_UUID_DECLARE_128(0xdb, 0x1f, 0xe2, 0x52, 0xf3, 0xc6, 0x43, 0x66, 0xb3, 0x92, 0x5d,      \
+			    0xc6, 0xe7, 0xc9, 0x59, 0x9d)
+
+#define UUID_2                                                                                     \
+	BT_UUID_DECLARE_128(0x3f, 0xa4, 0x7f, 0x44, 0x2e, 0x2a, 0x43, 0x05, 0xab, 0x38, 0x07,      \
+			    0x8d, 0x16, 0xbf, 0x99, 0xf1)
+
+static ssize_t read_mtu_validation_chrc(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+					void *buf, uint16_t buf_len, uint16_t offset)
+{
+	ssize_t read_len;
+
+	LOG_INF("Server side buf_len %u", buf_len);
+
+	/* Note: We assume `buf_len` is equal to the usable payload
+	 * capacity of the response PDU. I.e. `(ATT_MTU - 1)` for
+	 * BT_ATT_OP_READ_RSP and BT_ATT_OP_READ_BLOB_RSP.
+	 */
+
+	/* Send back a full PDU on the first read (on offset 0). Then an
+	 * not full one for the second read to conlude the long read..
+	 */
+	read_len = buf_len;
+	if (offset > 0) {
+		__ASSERT_NO_MSG(read_len > 0);
+		/* The second PDU is one-less-than-full to test for off
+		 * by one errors.
+		 */
+		read_len -= 1;
+	}
+
+	/* If the ATT_MTU is too large, sending a one-less-than-full
+	 * response would exeed the max attribute length limit.
+	 */
+	__ASSERT(buf_len < (BT_ATT_MAX_ATTRIBUTE_LEN / 2),
+		 "The EATT buffer is too large for this test.");
+
+	/* Ensure the padding bytes (that are not overwritten later in
+	 * this function) are initialized.
+	 */
+	memset(buf, 0, read_len);
+
+	/* Echo back the requested read size in the first two bytes of
+	 * each read.
+	 */
+	__ASSERT_NO_MSG(read_len >= 2);
+	sys_put_le16(read_len, buf);
+
+	return read_len;
+}
+
+static struct bt_gatt_attr attrs[] = {
+	BT_GATT_PRIMARY_SERVICE(UUID_1),
+	BT_GATT_CHARACTERISTIC(UUID_2, BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
+			       read_mtu_validation_chrc, NULL, NULL),
+};
+
+static struct bt_gatt_service svc = {
+	.attrs = attrs,
+	.attr_count = ARRAY_SIZE(attrs),
+};
+
+static void find_the_chrc(struct bt_conn *conn, uint16_t *chrc_value_handle)
+{
+	uint16_t svc_handle;
+	uint16_t svc_end_handle;
+	uint16_t chrc_end_handle;
+
+	EXPECT_ZERO(bt_testlib_gatt_discover_primary(&svc_handle, &svc_end_handle, conn, UUID_1, 1,
+						     0xffff));
+
+	LOG_INF("svc_handle: %u, svc_end_handle: %u", svc_handle, svc_end_handle);
+
+	EXPECT_ZERO(bt_testlib_gatt_discover_characteristic(chrc_value_handle, &chrc_end_handle,
+							    NULL, conn, UUID_2, (svc_handle + 1),
+							    svc_end_handle));
+
+	LOG_INF("chrc_value_handle: %u, chrc_end_handle: %u", *chrc_value_handle, chrc_end_handle);
+}
+
+static void bs_sync_all_log(char *log_msg)
+{
+	/* Everyone meets here. */
+	bt_testlib_bs_sync_all();
+
+	if (get_device_nbr() == 0) {
```

Comment:
```suggestion
	if (get_device_nbr() == CENTRAL_DEVICE_NBR) {
```

Reply:
This is not the intent of the code. It's not selecting the 'the central'. It's selecting 'device 0'.